internal/activities/verify_manifest_test.go

```diff
@@ -0,0 +1,136 @@
+package activities_test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	cp "github.com/otiai10/copy"
+	temporalsdk_activity "go.temporal.io/sdk/activity"
+	temporalsdk_testsuite "go.temporal.io/sdk/testsuite"
+	"gotest.tools/v3/assert"
+
+	"github.com/artefactual-sdps/preprocessing-sfa/internal/activities"
+	"github.com/artefactual-sdps/preprocessing-sfa/internal/sip"
+)
+
+func setup(t *testing.T) string {
+	t.Helper()
+
+	td := t.TempDir()
+
+	missingFile := filepath.Join(td, "missing_file")
+	err := cp.Copy("../testdata/little_digitized_AIP", missingFile)
+	if err != nil {
+		t.Fatalf("setup: copy: %v", err)
+	}
+
+	err = os.Remove(
+		filepath.Join(missingFile, "content", "content", "d_0000001", "00000001.jp2"),
+	)
+	if err != nil {
+		t.Fatalf("setup: remove: %v", err)
+	}
+
+	extraFile := filepath.Join(td, "extra_file")
+	err = cp.Copy("../testdata/little_digitized_AIP", extraFile)
+	if err != nil {
+		t.Fatalf("setup: copy: %v", err)
+	}
+
+	err = os.WriteFile(
+		filepath.Join(extraFile, "content", "content", "d_0000001", "extra_file.txt"),
+		[]byte("I'm an extra file."),
+		0o600,
+	)
+	if err != nil {
+		t.Fatalf("setup: write file: %v", err)
+	}
+
+	return td
+}
```

Comment:
It would be nice to fake the SIP structure and XML contents instead of having one in that testdata folder, it shouldn't be hard to do, there are some examples across the tests already:

https://github.com/artefactual-sdps/preprocessing-sfa/blob/main/internal/activities/transform_sip_test.go

I doesn't need to be a full SIP with the XSD files and all to prove the activity behavior.

Reply:
Yeah, good idea @jraddaoui.  I was originally going to check the file checksums too, which would require actual file contents to checksum, but after rereading the comments on #22 I noticed that we may be checking the metadata.xml checksums against the BagIt bag checksums after we generate the bag.  In any case, I think it's better to check the checksums in a separate activity.

TIL; I'll remove the test SIP and fake the XML + structure in the test file.